src/client/oicp/CpoOICPClient.ts

```diff
@@ -0,0 +1,1272 @@
+        message: 'Charging Station must be associated to a site area',
+        module: MODULE_NAME, method: 'updateEVSEStatus',
+      });
+    }
+    if (!chargingStation.issuer) {
+      throw new BackendError({
+        source: chargingStation.id,
+        action: ServerAction.OICP_UPDATE_EVSE_STATUS,
+        message: 'Only charging Station issued locally can be exposed to Hubject',
+        module: MODULE_NAME, method: 'updateEVSEStatus',
+      });
+    }
+    if (!chargingStation.public) {
+      throw new BackendError({
+        source: chargingStation.id,
+        action: ServerAction.OICP_UPDATE_EVSE_STATUS,
+        message: 'Private charging Station cannot be exposed to Hubject',
+        module: MODULE_NAME, method: 'updateEVSEStatus',
+      });
+    }
+    // Define get option
+    const options = {
+      addChargeBoxID: true,
+      countryID: this.getLocalCountryCode(ServerAction.OICP_PUSH_EVSE_STATUSES),
+      partyID: this.getLocalPartyID(ServerAction.OICP_PUSH_EVSE_STATUSES)
+    };
+    const evseStatus = OICPMapping.convertConnector2EvseStatus(this.tenant, chargingStation, connector, options);
+    const response = await this.pushEvseStatus([evseStatus], OICPActionType.update);
+    return response;
+  }
+
+  /**
+   * Push EVSE
+   */
+  async pushEvseData(evses: OICPEvseDataRecord[], actionType: OICPActionType): Promise<OICPAcknowledgment> {
+    let pushEvseDataResponse: OICPAcknowledgment = {} as OICPAcknowledgment;
+    // Check for input parameter
+    if (!evses) {
+      throw new BackendError({
+        action: ServerAction.OICP_PUSH_EVSES,
+        message: 'Invalid parameters',
+        module: MODULE_NAME, method: 'pushEvseData',
+      });
+    }
+    // Get EVSE endpoint url
+    // for QA - environment: https://service-qa.hubject.com/api/oicp/evsepush/v23/operators/{operatorID}/data-records
+    // for PROD - environment: https://service.hubject.com/api/oicp/evsepush/v23/operators/{operatorID}/data-records
+    const fullUrl = this.getEndpointUrl('evses', ServerAction.OICP_PUSH_EVSES);
+
+    // Build payload
+    const operatorEvseData: OICPOperatorEvseData = {} as OICPOperatorEvseData;
+    operatorEvseData.OperatorID = this.getOperatorID(ServerAction.OICP_PUSH_EVSES);
+    operatorEvseData.OperatorName = this.tenant.name;
+    operatorEvseData.EvseDataRecord = evses;
+
+    const payload: OICPPushEvseDataCpoSend = {} as OICPPushEvseDataCpoSend;
+    payload.ActionType = actionType;
+    payload.OperatorEvseData = operatorEvseData;
+
+    // Log
+    Logging.logDebug({
+      tenantID: this.tenant.id,
+      action: ServerAction.OICP_PUSH_EVSES,
+      message: `Push EVSEs from tenant: ${this.tenant.id}`,
+      module: MODULE_NAME, method: 'pushEvseData',
+      detailedMessages: { payload }
+    });
+
+    // Call Hubject
+    await this.axiosInstance.post(fullUrl, payload
+    ).then(function(response) {
+      console.log('Success! pushEvseData: ', response);
```

Comment:
Remove debug code or use debug logging method.

Reply:
Yes, I will remove it in the future.
At the moment I need the console logs.